src/test/java/gov/healthit/chpl/aqa/stepDefinitions/LoginSteps.java

```diff
@@ -14,50 +13,73 @@
+        if (StringUtils.isEmpty(roleAdminUsername)) {
+            throw new IllegalArgumentException("Missing value for roleAdminUsername!");
         }
-
-        if (StringUtils.isEmpty(username)) {
-            throw new IllegalArgumentException("Missing value for username!");
+        if (StringUtils.isEmpty(roleAdminPassword)) {
+            throw new IllegalArgumentException("Missing value for roleAdminPassword!");
+        }
+        if (StringUtils.isEmpty(roleOncUsername)) {
+            throw new IllegalArgumentException("Missing value for roleOncUsername!");
+        }
+        if (StringUtils.isEmpty(roleOncPassword)) {
+            throw new IllegalArgumentException("Missing value for roleOncPassword!");
+        }
+        if (StringUtils.isEmpty(roleAcbUsername)) {
+            throw new IllegalArgumentException("Missing value for roleAcbUsername!");
         }
-        if (StringUtils.isEmpty(password)) {
-            throw new IllegalArgumentException("Missing value for password!");
+        if (StringUtils.isEmpty(roleAcbPassword)) {
+            throw new IllegalArgumentException("Missing value for roleAcbPassword!");
         }
     }
 
     /**
-     * Log the user in as an ACB.
+     * Verify login attempt was successful.
+     * @param name as AQA Admin, AQA ONC or AQA ACB
      */
-    @Given("^I'm logged in as an ACB$")
-    public void logInAsAcb() {
-        driver.get(url + "#/admin/dpManagement");
-        WebDriverWait wait = new WebDriverWait(driver, TIMEOUT);
-        wait.until(ExpectedConditions.visibilityOf(LoginPage.username(driver)));
-        LoginPage.username(driver).sendKeys(username);
-        LoginPage.password(driver).sendKeys(password);
-        LoginPage.loginButton(driver).click();
+    @Then("^I should be logged in to CHPL as \"([^\"]*)\"$")
+    public void verifyLoginWasSuccessful(final String name) {
+        String actualString = LoginPage.welcomeText(getDriver()).getText();
+        assertTrue(actualString.contains(name));
     }
 
     /**
-     * Verify login attempt was successful.
+     * Log the user in as given ROLE.
+     * @param role as ROLE_ADMIN, ROLE_ONC or ROLE_ACB
      */
-    @Then("^I should be logged in to CHPL as ACB Admin$")
-    public void verifyLoginWasSuccessful() {
-        String actualString = LoginPage.welcomeText(driver).getText();
-        assertTrue(actualString.contains("Welcome"));
+    @Given("^I'm logged in as \"([^\"]*)\"$")
+    public void logInAsRole(final String role) {
+        String username = null;
+        String password = null;
+        if (role.equalsIgnoreCase("ROLE_ADMIN")) {
+            username = roleAdminUsername;
+            password = roleAdminPassword;
+        } else if (role.equalsIgnoreCase("ROLE_ONC")) {
+            username = roleOncUsername;
+            password = roleOncPassword;
+        } else if (role.equalsIgnoreCase("ROLE_ACB")) {
+            username = roleAcbUsername;
+            password = roleAcbPassword;
+        }
```

Comment:
```suggestion
        } else {
        throw new IllegalArgumentException("Incorrect role found: " + role);
        }
```
If you don't catch this here, we'd just end up waiting for the login page to load, then waiting for an empty username/password to be submitted, which probably wouldn't even work, but in any case we should do something about it